addons/udes_report/models/stock_export.py

```diff
@@ -62,79 +62,103 @@ def run_stock_file_export(self, context=None, send_file_via='user'):
             1) part number, location, package ID and quantity;
             2) part number, number of packages and total quantity.
         '''
-        locations = self.included_locations - self.excluded_locations
+        file_name = 'warehouse_stock_{TIMESTAMP}.xls'.format(TIMESTAMP=datetime.now())
 
-        if not locations:
-            raise UserError(_("The specified list of Stock Locations is empty."))
+        data = self._generate_stock_file_data()
+        self._write_workbook(file_name, "Stock File", data, send_file_via=send_file_via)
 
+    def _generate_stock_file_data(self):
+        '''
+        Generate data for the stock file.
+
+        This will contain 2 sheets that may then be exported to any format
+        (currently only excel)
+        Sheets:
+            1) part number, location, package ID and quantity;
+            2) part number, number of packages and total quantity.
+        '''
         Quant = self.env['stock.quant']
         Product = self.env['product.product']
         Location = self.env['stock.location']
-        all_quants = Quant.search([('location_id', 'child_of', locations.ids)])
-        by_prod = lambda x: x.product_id.id
-        by_location = lambda x: x.location_id.id
-        by_package_name = lambda x: x.package_id.name or ''
 
-        get_quants_by_loc = lambda quant_union: {
-            Location.browse(loc): Quant.union(*_quants)
-            for loc, _quants in groupby(quant_union.sorted(key=by_location),
-                                        key=by_location)}
+        def by_prod(record):
+            return record.product_id.id
+
+        def by_location(record):
+            return record.location_id.id
+
+        def by_package_name(record):
+            return record.package_id.name or ''
+
+        def by_lot_name(record):
+            return record.lot_id.name or ''
+
+        def get_quants_by_loc(quant_union):
+            return {Location.browse(loc): Quant.union(*_quants)
+                    for loc, _quants in groupby(quant_union.sorted(key=by_location),
+                                                key=by_location)}
+
+        locations = self.included_locations - self.excluded_locations
+        if not locations:
+            raise UserError(_("The specified list of Stock Locations is empty."))
+
+        all_quants = Quant.search([('location_id', 'child_of', locations.ids)])
 
         quants_by_prod_by_loc = OrderedDict()
         prod_summary = defaultdict(lambda: {'n_pkgs': 0, 'qty': 0})
 
-        for prod, _quants in groupby(all_quants.sorted(key=by_prod),
-                                     key=by_prod):
+        # Determine if lots/serials should be shown on the report
+        show_tracking = bool(Product.search([("tracking", "!=", "none")]))
```

Comment:
Use `search_count` here instead of `search`.

Reply:
Changed